recipes/pybind11_json/all/conanfile.py

```diff
@@ -23,8 +23,8 @@ def layout(self):
         basic_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("nlohmann_json/3.11.2")
-        self.requires("pybind11/2.10.4")
+        self.requires("nlohmann_json/[>=3.9 <4]")
```

Comment:
I have checked the occurences for this in ConanCenter.
There are like 38 of them, the vast majority to ``3.11.3``. This is also the latest version on ConanCenter.
So the version range wouldn't solve much, using versions like 3.9 is a bit untested, and allowing newer versions doesn't change much if the latest is 3.11.3.

We also did some research, to propose version ranges for ``nlohmann_json``, but we didn't find any evidence (not favor not against) the binary compatibility of the library, so we didn't decide to move forward with it yet.

If the previous ``3.11.2`` was raising conflicts, maybe updating it to ``3.11.3`` would be enough to solve the conflicts?

Reply:
My reasoning for choosing a range here was that `nlohmann_json` is a header-only library, as is pybind11 and this recipe, so there is no ABI to worry about to begin with.

Conan does not appear to have a system in place yet to ensure that recipe dependencies do not become stale, so I'd prefer ranges whenever possible, otherwise recipes like this one become unavailable when the target application requires a more recent dependency of this recipe as well. For pybind11 and nlohmann_json, that is bound to happen very frequently.